charts/vela-minimal/templates/defwithtemplate/gateway.yaml

```diff
@@ -107,31 +97,51 @@ spec:
 
         	// +usage=Specify the host of the ingress gateway, which is used to generate the endpoints when the host is empty.
         	gatewayHost?: string
+
+        	// +usage=Specify a unique name for this gateway, required to support multiple gateway traits on a component
+        	name?: string
+
+        	// +usage=Specify a pathType for the ingress rules, defaults to "ImplementationSpecific"
+        	pathType: *"ImplementationSpecific" | "Prefix" | "Exact"
         }
   status:
     customStatus: |-
-      if context.outputs.ingress.status.loadBalancer.ingress == _|_ {
+      let nameSuffix = {
+        if parameter.name != _|_ { "-" + parameter.name }
+        if parameter.name == _|_ { "" }
+      }
+      let ingressMetaName = context.name + nameSuffix
+      let ig  = [for i in context.outputs if (i.kind == "Ingress") && (i.metadata.name == ingressMetaName) {i}][0]
+      let igs = ig.status.loadBalancer.ingress[0]
+      let igr = ig.spec.rules[0]
+      if igs == _|_ {
         message: "No loadBalancer found, visiting by using 'vela port-forward " + context.appName + "'\n"
       }
-      if context.outputs.ingress.status.loadBalancer.ingress != _|_ {
-      	let igs = context.outputs.ingress.status.loadBalancer.ingress
-      	let host = context.outputs.ingress.spec.rules[0].host
-        if igs[0].ip != _|_ {
-        	if host != _|_ {
-      	    message: "Visiting URL: " + context.outputs.ingress.spec.rules[0].host + ", IP: " + igs[0].ip
-        	}
-        	if host == _|_ {
-      	    message: "Host not specified, visit the cluster or load balancer in front of the cluster with IP: " + igs[0].ip
-        	}
+      if igs != _|_ {
+        if igs.ip != _|_ {
```

Comment:
You can use double if to simplify like:

```
if igs != _|_ if igs.ip != _|_ {
}
```

Reply:
Yeah I think it's good with individual lines of ifs as long as we're not using `&&` for short-circuit.